### PR TITLE
Allows proper discovery of app frontend configurations #11253

### LINKED ISSUE #11253: Allow proper discovery of Arches applications in frontend

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

Closes #11253 

### Checklist

-   I targeted one of these branches:
    - [x] dev/7.6.x (under development): features, bugfixes not covered below
    - [ ] dev/7.5.x (main support): regressions, crashing bugs, security issues, major bugs in new features
    - [ ] dev/6.2.x (extended support): major security issues, data loss issues
-   [ ] I added a changelog in arches/releases
-   [ ] I submitted a PR to arches-docs (if appropriate)
-   [ ] Unit tests pass locally with my changes
-   [ ] I added tests that prove my fix is effective or that my feature works
-   [ ] My test fails on the target branch

#### Accessibility Checklist

[Developer Guide](https://arches.readthedocs.io/en/stable/developing/advanced/accessibility/)

| Topic            | Changed | Retested |
| ---------------- | ------- | -------- |
| Color contrast   |         |          |
| Form fields      |         |          |
| Headings         |         |          |
| Links            |         |          |
| Keyboard         |         |          |
| Responsive Design|         |          |
| HTML validation  |         |          |
| Screen reader    |         |          |

#### Ticket Background
*   Sponsored by: 
*   Found by: @ 
*   Tested by: @ 
*   Designed by: @

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
@chrabyrd Just gave this a spin with `arches-references`. I ran `updateproject` and got the new changes in apps.py. Let me know if I missed any other steps.

**Result:**

- I can now change the `arches/arches/app/src/arches/types` paths back to `@/arches...`
- but I now see failures with the `@/arches-references` alias, and full-pathing it doesn't seem to work (neither does changing types -> types.ts)

See [example failure](https://github.com/archesproject/arches-references/pull/20/commits/f40686fd07eb70eda9fac231927d33981fdd0e0c)
***
The code in this PR looks really polished, I only have some minor comments mostly on the Django side, but I'll hold them until we know this aliasing piece works as intended though.

**Comment by jacobtylerwalls** [comment]
The relevant failures look like this:

```
arches_references/src/arches-references/utils.ts(9,8): error TS2307: Cannot find module '@/arches-references/types' or its corresponding type declarations.
```

and then some stuff downstream of that. Pretty similar to the case this PR fixes for core arches, described in https://github.com/archesproject/arches/issues/11234#issue-2426202586.

**Review by jacobtylerwalls — COMMENTED** [review]
Thanks for pitching in to debug this. Some minor comments from a code read.

File: arches/install/arches-templates/vitest.config.mts [hunk 1]
```diff
@@ -1,36 +1,64 @@
...
+            '@/arches': path.join(parsedData['ROOT_DIR'], 'app', 'src', 'arches'),
+        };
+
+        for (
+            const [archesApplicationName, archesApplicationPath] 
+            of Object.entries(
+                parsedData['ARCHES_APPLICATIONS_PATHS'] as { [key: string]: string }
+            )
+        ) {
+            alias[path.join('@', archesApplicationName)] = path.join(archesApplicationPath, 'src', archesApplicationName);
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply arches/install/arches-templates/vitest.config.mts:34]: good 🎣

File: arches/install/arches-templates/webpack/webpack.common.js [hunk 2]
```diff
@@ -8,551 +8,509 @@ const MiniCssExtractPlugin = require('mini-css-extract-plugin');
...
+        // END create vue filepath lookup
+        // BEGIN create universal constants
 
-            // END create universal constants
+        const universalConstants = {
+            APP_ROOT_DIRECTORY: JSON.stringify(APP_ROOT).replace(/\\/g, '/'),
+            ARCHES_CORE_DIRECTORY: JSON.stringify(ROOT_DIR).replace(/\\/g, '/'),
+            ARCHES_APPLICATIONS: JSON.stringify(ARCHES_APPLICATIONS),
+            SITE_PACKAGES_DIRECTORY: JSON.stringify(SITE_PACKAGES_DIRECTORY).replace(/\\/g, '/'),
+        };
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply arches/install/arches-templates/webpack/webpack.common.js:248]: ya definitely worth making an exploratory ticket though 👍

File: arches/management/commands/run_ready.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# pragma: no cover
+
+from django.core.management.base import BaseCommand
+from django.apps import apps
+from django.conf import settings
+
+
+class Command(BaseCommand):
+    help = "Runs the ready method of the project-level AppConfig"
```
[the remark this answers is not in the document]
Comment (chrabyrd) [reply arches/management/commands/run_ready.py:9]: !!!

TIL

**Comment by jacobtylerwalls** [comment]
Typescript failures on https://github.com/archesproject/arches-references/pull/20 now look legit (or symptoms of no-longer-needed workarounds).

**Review by jacobtylerwalls — APPROVED** [review]
+1 tested, thanks for the updates!

**Pull request merged**